funds-manager/funds-manager-server/src/execution_client/mod.rs

```diff
@@ -61,4 +80,12 @@ impl ExecutionClient {
             .await
             .map_err(ExecutionClientError::http)
     }
+
+    /// Get an instance of a signer middleware with the http provider attached
+    fn get_signer(
+        &self,
+        wallet: LocalWallet,
+    ) -> SignerMiddleware<Arc<Provider<Http>>, LocalWallet> {
+        SignerMiddleware::new(self.rpc_provider.clone(), wallet)
```

Comment:
we could attach the chain ID here for convenience

Reply:
this is done in a follow up on the wallet directly